components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -10557,6 +10557,69 @@ public static Map getTenantBasedStoreDomainMapping(String tenantDomain) throws A
         return null;
     }
 
+    public static Map getTenantBasedPublisherDomainMapping(String tenantDomain) throws APIManagementException {
+
+        try {
+            Registry registry = ServiceReferenceHolder.getInstance().getRegistryService().
+                    getGovernanceSystemRegistry();
+            String resourcePath = APIConstants.API_DOMAIN_MAPPINGS.replace(APIConstants.API_DOMAIN_MAPPING_TENANT_ID_IDENTIFIER, tenantDomain);
+            if (registry.resourceExists(resourcePath)) {
+                Resource resource = registry.get(resourcePath);
+                String content = new String((byte[]) resource.getContent(), Charset.defaultCharset());
+                JSONParser parser = new JSONParser();
+                JSONObject mappings = (JSONObject) parser.parse(content);
+                if (mappings.containsKey(APIConstants.API_DOMAIN_MAPPINGS_PUBLISHER)) {
+                    return (Map) mappings.get(APIConstants.API_DOMAIN_MAPPINGS_PUBLISHER);
+                }
+            }
+        } catch (RegistryException e) {
+            String msg = "Error while retrieving publisher domain mappings from registry";
+            throw new APIManagementException(msg, e);
+        } catch (ClassCastException e) {
```

Comment:
Better not to catch ClassCastException. Ideally, you should check the instance type and continue.